### PR TITLE
feat: chore(apig): support APPCODE resource and update the old parameter

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
Support a resource to manage APPCODE.
Update the parameter `app_codes` of the application resource and ignore changes.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. support the APPCODE resource for APIG service.
2. support related documentation and acceptance test.
3. update parameter `app_codes` behavior.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

1. Auto generate APPCODE test
```
make testacc TEST='./huaweicloud/services/acceptance/apig' TESTARGS='-run=TestAccAppcode_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/apig -v -run=TestAccAppcode_basic -timeout 360m -parallel 4
=== RUN   TestAccAppcode_basic
=== PAUSE TestAccAppcode_basic
=== CONT  TestAccAppcode_basic
--- PASS: TestAccAppcode_basic (465.60s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/apig    465.666s
```
2. Manually configure APPCODE test
```
make testacc TEST='./huaweicloud/services/acceptance/apig' TESTARGS='-run=TestAccAppcode_manuallyConfig'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/apig -v -run=TestAccAppcode_manuallyConfig -timeout 360m -parallel 4
=== RUN   TestAccAppcode_manuallyConfig
=== PAUSE TestAccAppcode_manuallyConfig
=== CONT  TestAccAppcode_manuallyConfig
--- PASS: TestAccAppcode_manuallyConfig (471.79s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/apig      471.891s
```
3. Manually destroy and refresh state
销毁并刷新
4. Manually destroy when destroying APPCODE and before confirm
删除时销毁

### CODE REVIEW

File: huaweicloud/services/apig/resource_huaweicloud_apig_appcode.go [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+	)
+	client, err := cfg.ApigV2Client(region)
+	if err != nil {
+		return diag.Errorf("error creating APIG v2 client: %s", err)
+	}
+
+	resp, err := applications.GetAppCode(client, instanceId, appId, appCodeId).Extract()
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, fmt.Sprintf("error querying APPCODE (%s) from specified application (%s)",
+			appCodeId, appId))
```
Comment (hmbig2) [inline huaweicloud/services/apig/resource_huaweicloud_apig_appcode.go]: add instanceId ?

File: docs/resources/apig_application_code.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
+---
+subcategory: "API Gateway (Dedicated APIG)"
+---
+
+# huaweicloud_apig_appcode
+
+Manages an APPCODE in application resource within HuaweiCloud.
```
Comment (deer-hang) [inline docs/resources/apig_application_code.md:7]:
```suggestion
Manages an APIG APPCODE in application resource within HuaweiCloud.
```

File: huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_appcode_test.go [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+		rName := "huaweicloud_apig_appcode.test"
+		rs, ok := s.RootModule().Resources[rName]
+		if !ok {
+			return "", fmt.Errorf("Resource (%s) not found: %s", rName, rs)
+		}
+		if rs.Primary.ID == "" || rs.Primary.Attributes["instance_id"] == "" || rs.Primary.Attributes["application_id"] == "" {
+			return "", fmt.Errorf("invalid format specified for import ID, want '<instance_id>/<application_id>/<policy_id>', but got '%s/%s/%s'",
+				rs.Primary.Attributes["instance_id"], rs.Primary.Attributes["application_id"], rs.Primary.ID)
+		}
+		return fmt.Sprintf("%s/%s/%s", rs.Primary.Attributes["instance_id"], rs.Primary.Attributes["application_id"], rs.Primary.ID), nil
```
Comment (deer-hang) [inline huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_appcode_test.go]: Define variables for repeatedly referenced attributes.
```
id := rs.Primary.ID
instanceId := rs.Primary.Attributes["instance_id"]
applicationId := rs.Primary.Attributes["application_id"]
```

File: huaweicloud/services/apig/resource_huaweicloud_apig_appcode.go [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+	client, err := cfg.ApigV2Client(region)
+	if err != nil {
+		return diag.Errorf("error creating APIG v2 client: %s", err)
+	}
+
+	err = applications.RemoveAppCode(client, instanceId, appId, appCodeId).ExtractErr()
+	if _, ok := err.(golangsdk.ErrDefault404); ok || err == nil {
+		return nil
+	}
+	return diag.Errorf("error deleting APPCODE (%s) from the specified application (%s): %s", appCodeId, appId, err)
```
Comment (deer-hang) [inline huaweicloud/services/apig/resource_huaweicloud_apig_appcode.go]:
```suggestion
	if err != nil {
		if _, ok := err.(golangsdk.ErrDefault404); ok {
			return nil
		}
		return diag.Errorf("error deleting APPCODE (%s) from the specified application (%s): %s", appCodeId, appId, err)
	}
	return nil
```

File: huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_appcode_test.go (hunk 3 above)
Comment (deer-hang) [reply huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_appcode_test.go]: The log message `<policy_id>` should be `<id>`

File: huaweicloud/services/apig/resource_huaweicloud_apig_appcode.go (hunk 1 above)
Comment (Lance52259) [reply huaweicloud/services/apig/resource_huaweicloud_apig_appcode.go]: okey

File: huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_appcode_test.go (hunk 3 above)
Comment (Lance52259) [reply huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_appcode_test.go]: fixed.

File: huaweicloud/services/apig/resource_huaweicloud_apig_appcode.go (hunk 4 above)
Comment (Lance52259) [reply huaweicloud/services/apig/resource_huaweicloud_apig_appcode.go]: the original logic is clearer.
And in there, maybe `CheckDeletedDiag()` is better

Comment (Lance52259) [reply huaweicloud/services/apig/resource_huaweicloud_apig_appcode.go]: refactored

File: huaweicloud/services/apig/resource_huaweicloud_apig_appcode.go [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+	if err != nil {
+		return common.CheckDeletedDiag(d, err,
+			fmt.Sprintf("error deleting APPCODE (%s) from specified application (%s) under dedicated instance (%s)",
+				appCodeId, appId, instanceId))
+	}
+	return nil
+}
+
+func resourceAppcodeImportState(_ context.Context, d *schema.ResourceData,
+	_ interface{}) ([]*schema.ResourceData, error) {
```
Comment (ShiChangkuo) [inline huaweicloud/services/apig/resource_huaweicloud_apig_appcode.go]:
```suggestion
func resourceAppcodeImportState(_ context.Context, d *schema.ResourceData, _ interface{}) ([]*schema.ResourceData, error) {
```

Comment (Lance52259) [reply huaweicloud/services/apig/resource_huaweicloud_apig_appcode.go]:
```
func resourceAppcodeImportState(_ context.Context, d *schema.ResourceData, _ interface{}) ([]*schema.ResourceData,
    error) {
```
124 characters in this line.

**Pull request merged**